Fix PATH in convenience scripts

Add the convenience script's directory to PATH, on the assumption that
the multimarkdown executable is in the same directory.

Remove $PWD from PATH, because there's no reason why we should expect
the user's shell to have been changed to the directory that contains
multimarkdown.

export PATH separately from setting PATH, because older shells can't do
both at once.

unset CDPATH because if a user had exported it, it could change the
behavior of cd.

https://github.com/fletcher/MultiMarkdown-6/pull/148
--- scripts/markdown.orig
+++ scripts/markdown
@@ -8,7 +8,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd.orig
+++ scripts/mmd
@@ -8,7 +8,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd2all.orig
+++ scripts/mmd2all
@@ -9,7 +9,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd2epub.orig
+++ scripts/mmd2epub
@@ -1,6 +1,6 @@
 #!/bin/sh
 #
-# mmd2oepub --- MultiMarkdown convenience script
+# mmd2epub --- MultiMarkdown convenience script
 #	<http://fletcherpenney.net/multimarkdown/>
 #	Fletcher T. Penney
 #
@@ -8,7 +8,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd2fodt.orig
+++ scripts/mmd2fodt
@@ -8,7 +8,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd2odt.orig
+++ scripts/mmd2odt
@@ -8,7 +8,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd2opml.orig
+++ scripts/mmd2opml
@@ -8,7 +8,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd2pdf.orig
+++ scripts/mmd2pdf
@@ -15,7 +15,17 @@
 #		LaTeX file.
 
 # Be sure to include multimarkdown and latex in our PATH
-export PATH="$PWD:/usr/local/bin:/usr/texbin:/Library/TeX/texbin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:/usr/texbin:/Library/TeX/texbin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
--- scripts/mmd2tex.orig
+++ scripts/mmd2tex
@@ -8,7 +8,17 @@
 #
 
 # Be sure to include multimarkdown in our PATH
-export PATH="$PWD:/usr/local/bin:$PATH"
+unset CDPATH
+SCRIPT="$0"
+while
+	DIR="$(cd -P -- "$(dirname -- "$SCRIPT")" && pwd)"
+	test -h "$SCRIPT"
+do
+	SCRIPT="$(readlink "$SCRIPT")"
+	[ "$SCRIPT" = "${SCRIPT#/}" ] && SCRIPT="$DIR/$SCRIPT"
+done
+PATH="$DIR:/usr/local/bin:$PATH"
+export PATH
 
 which multimarkdown > /dev/null
 if [ $? = 1 ]
